### PR TITLE
Typo Corrections and Doc Fixes

### PR DESCRIPTION
**Pull request by jigsaw004**
### Description of changes:
Typo corrections and formatting fixes.

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/analysis.md [hunk 1]
```diff
@@ -243,11 +243,11 @@ trial.collections()
 ```
 
 ###### Returns
-`dict[str -> Collection]` A dictionary indexed by the name of the collection, with the Collection object as the value. Please refer [Collection API](api.md) for more details. #TODO fix link
+`dict[str -> Collection]` A dictionary indexed by the name of the collection, with the Collection object as the value. Please refer [Collection API](https://github.com/awslabs/sagemaker-debugger/blob/master/docs/api.md) for more details.
```
Comment (vandanavk) [inline docs/analysis.md]: Could you point to this https://github.com/awslabs/sagemaker-debugger/blob/master/docs/api.md#collection

Comment (jigsaw004) [reply docs/analysis.md]: Done.

File: docs/analysis.md [hunk 2]
```diff
@@ -478,38 +509,38 @@ invoke_rule(rule_obj, start_step=0, end_step=None)
...
 
 ```python
 from smdebug.exceptions import *
 ```
 
-Here are the exceptions and their meanings:
+Here are the exceptions (along with others) and their meaning:
 
-- `TensorUnavailableForStep` : This means that the tensor requested is not available for the step. It may be saved for a different step number. Note that when you see this exception, it means that this tensor can never become available for this step in the future.
+- `TensorUnavailableForStep` : This means that the tensor requested is not available for the step. It may have been saved or will be saved for a different step number. Note that this exception implies that the requested tensor will never become available for this step in the future.
```
Comment (Vikas-kum) [inline docs/analysis.md]: ==>"will be saved for a different step number" 
Not needed. 
"saved for a different step number. " -> "saved for a different step number. You can check which steps tensor is saved for by trial.tensor('tname').steps() api [https://github.com/awslabs/sagemaker-debugger/blob/master/docs/analysis.md#steps-1]"

File: docs/analysis.md [hunk 3]
```diff
@@ -122,26 +122,26 @@ trial.tensor_names(step= None,
...
-- `trial.tensor_names(collection='gradients')` Returns tensors from gradients collection
-- `trial.tensor_names(step=10, mode=modes.TRAIN, regex='softmax')` Returns tensor saved for 10th training step which match the regex `softmax`
+- `trial.tensor_names(regex='relu')` Returns all tensors matching the regex pattern `relu` saved for any step or mode.
+- `trial.tensor_names(collection='gradients')` Returns tensors from collection "gradients"
+- `trial.tensor_names(step=10, mode=modes.TRAIN, regex='softmax')` Returns tensor saved for 10th training step which matches the regex `softmax`
 
 
 #### tensor
-Retrieve the `smdebug.core.tensor.Tensor` object by the given name tname. You can review all the methods that this Tensor object provides [here](#Tensor).
+Retrieve the `smdebug.core.tensor.Tensor` object by the given name `tname`. You can review all the methods that this Tensor object provides [here](#Tensor).
```
Comment (Vikas-kum) [inline docs/analysis.md]: can you correct the link? #Tensor-1 , and also at line 152

File: docs/analysis.md [hunk 4]
```diff
@@ -51,10 +51,10 @@ Trial is capable of loading new tensors as and when they become available at the
...
 
 #### Non SageMaker training jobs
-If you are not running a SageMaker training job, this is the path you pass as out_dir when you create a smdebug [`Hook`](hook.md). Just like when creating the hook, you can pass either a local path or an S3 path (as `s3://bucket/prefix`).
+If you are not running a SageMaker training job, this is the path you pass as `out_dir` when you create a smdebug [`Hook`](hook.md). Just like when creating the hook, you can pass either a local path or an S3 path (as `s3://bucket/prefix`).
```
Comment (Vikas-kum) [inline docs/analysis.md]: link here should be (api.md#hook)

File: docs/analysis.md [hunk 5]
```diff
@@ -415,17 +415,17 @@ trial.tensor(name).prev_steps(step, n, mode=modes.GLOBAL)
...
+Rules are the medium by which SageMaker Debugger executes a certain piece of code regularly on different steps of a training job. A rule is assigned to a trial and can be invoked at each new step of the trial. It can also access other trials for its evaluation. You can evaluate a rule using tensors from the current step or any step before the current step. Please ensure your logic respects these semantics, else you will get a `TensorUnavailableForStep` exception as the data would not yet be available for future steps.
 
 ### Built In Rules
 Please refer to the built-in rules that SageMaker provides [here](https://github.com/awslabs/sagemaker-debugger/blob/master/docs/sagemaker.md#built-in-rules).
 
 ### Writing a custom rule
-Writing a rule involves implementing the [Rule interface](../../smdebug/rules/rule.py). Below let us look at a simplified version of a VanishingGradient rule.
+Writing a rule involves implementing the [Rule interface](../../smdebug/rules/rule.py). Below, let us look at a simplified version of a VanishingGradient rule.
```
Comment (Vikas-kum) [inline docs/analysis.md]: link needs a fix: https://github.com/awslabs/sagemaker-debugger/blob/master/smdebug/rules/rule.py 

../../smdebug/rules/rule.py --> ../smdebug/rules/rule.py

File: docs/analysis.md [hunk 6]
```diff
@@ -460,10 +459,42 @@ A simplified version of the actual invoke function for `VanishingGradientRule` i
...
+			"rule_to_invoke": "VanishingGradient",
+			"threshold": "20.0"
+		}
+	}
+]
+```
 
-#### invoke_rule
+#### Invoking a rule outside of SageMaker through `invoke_rule`
 You might want to invoke the rule locally during development. We provide a function to invoke rules easily. Refer [smdebug/rules/rule_invoker.py](../../smdebug/rules/rule_invoker.py). The invoke function has the following syntax. It takes a instance of a Rule and invokes it for a series of steps one after the other.
```
Comment (Vikas-kum) [inline docs/analysis.md]: link fix ,
../../smdebug/rules/rule_invoker.py -->../smdebug/rules/rule_invoker.py
